### ISSUE TITLE
Add unchecked_pow?

### ISSUE THREAD

**Issue by PallHaraldsson** [JuliaLang/julia#53297]
Do do have Base.Checked.checked_pow (seemingly starting in 1.11), and e.g.

Base.Checked.add_with_overflow
Base.Checked.sub_with_overflow
Base.Checked.mul_with_overflow

[which I'm not sure what are for, similar needed for pow?]

I don't like personally ^changed to use Base.Checked.checked_pow, but if don (now or) later, do we need unchecked_pow? I think there's no harm in having it, and it's a trivial (redundant) addition, and I think we should add it for 1.11, i.e. before the feature freeze (or ok later since trivial?).

My preference would be a Float64 result... then having unchecked_pow is also of potential help, i.e. either way, though the name a bit odd, so maybe just add this as `pow`?

**Comment by adienes** [comment]
`^` is unchecked_pow

**Comment by PallHaraldsson** [comment]
>^ is unchecked_pow

Yes, I mean still add it redundantly so that ^could later be changed to checked pow. Doesn't need to happen right away; and also other types like hypothetical type CInt could be checked using regular ^always sooner and use unchecked_pow when you know unwanted.
